### PR TITLE
Truncate float to integer conversion

### LINKED ISSUE #419: Allow as_type to convert from float to integer
I think our current implementation is too restrictive. I have an implementation that ends up with `f32` tensors:

```
rows = Nx.tensor([0.45, 0.52, 0.7])
```

And I need to use them to index:

```
indices = Nx.floor(rows) |> Nx.as_type({:s, 64})
```

But because of the restriction the current implementation is impossible

### PR DESCRIPTION
**Pull request by seanmor5**
Resolves #419

### CODE REVIEW

**Comment by seanmor5** [comment]
Agh, I'm working on Windows and the formatter does not seem to be working correctly (something with line endings I think?).

File: nx/lib/nx/binary_backend.ex [hunk 1]
```diff
@@ -2,7 +2,7 @@ defmodule Nx.BinaryBackend do
   @moduledoc """
   An opaque backend written in pure Elixir that stores
   the data in Elixir's binaries.
-
+
```
Comment (jonatanklosko) [inline nx/lib/nx/binary_backend.ex:5]: You have a trailing whitespace here, that's what the formatter complains about :)

Comment (seanmor5) [inline nx/lib/nx/binary_backend.ex:5]:
```suggestion

```

**Review by josevalim — APPROVED** [review]

File: nx/lib/nx/binary_backend.ex [hunk 2]
```diff
@@ -2,7 +2,6 @@ defmodule Nx.BinaryBackend do
   @moduledoc """
   An opaque backend written in pure Elixir that stores
   the data in Elixir's binaries.
-
   This is the default backend used by the `Nx` module.
```
Comment (jonatanklosko) [inline nx/lib/nx/binary_backend.ex:5]:
```suggestion

  This is the default backend used by the `Nx` module.
```

**Pull request merged**